great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -202,41 +242,77 @@ def get_effective_column_names(
-    ) -> List[Domain]:
-        """
-        Obtains and returns domains for all columns of a table (or for configured columns, if they exist in the table).
-        """
-        batch_ids: List[str] = self.get_batch_ids(variables=variables)
-
-        validator: "Validator" = self.get_validator(variables=variables)  # noqa: F821
+        # include_column_name_suffixes column_name_suffixes from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        include_column_name_suffixes: Optional[
+            Union[str, Iterable, List[str]]
+        ] = get_parameter_value_and_validate_return_type(
+            domain=None,
+            parameter_reference=self.include_column_name_suffixes,
+            expected_return_type=None,
+            variables=variables,
+            parameters=None,
+        )
 
-        table_column_names: List[str] = self.get_effective_column_names(
-            batch_ids=batch_ids,
-            validator=validator,
+        # exclude_column_name_suffixes column_name_suffixes from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        exclude_column_name_suffixes: Optional[
+            Union[str, Iterable, List[str]]
+        ] = get_parameter_value_and_validate_return_type(
+            domain=None,
+            parameter_reference=self.exclude_column_name_suffixes,
+            expected_return_type=None,
             variables=variables,
+            parameters=None,
         )
 
+        if include_column_name_suffixes:
+            if isinstance(include_column_name_suffixes, str):
+                include_column_name_suffixes = [include_column_name_suffixes]
+            else:
+                if not isinstance(include_column_name_suffixes, (Iterable, list)):
+                    raise ValueError(
+                        "Unrecognized include_column_name_suffixes directive -- must be a list or a string."
+                    )
+
+            effective_column_names: List[str] = list(
+                filter(
+                    lambda candidate_column_name: candidate_column_name.endswith(
+                        tuple(include_column_name_suffixes)
+                    ),
+                    effective_column_names,
+                )
+            )
+
+        if exclude_column_name_suffixes:
+            if isinstance(exclude_column_name_suffixes, str):
+                exclude_column_name_suffixes = [exclude_column_name_suffixes]
+            else:
+                if not isinstance(exclude_column_name_suffixes, (Iterable, list)):
+                    raise ValueError(
+                        "Unrecognized exclude_column_name_suffixes directive -- must be a list or a string."
+                    )
+
+            effective_column_names: List[str] = list(
+                filter(
+                    lambda candidate_column_name: not candidate_column_name.endswith(
+                        tuple(include_column_name_suffixes)
```

Comment:
Should this instead be 
```suggestion
                        tuple(exclude_column_name_suffixes)
```

Reply:
Fixed -- and committed -- and added a test -- thanks so much, @anthonyburdi !